toolkit/components/uniffi-bindgen-gecko-js/src/templates/js/Strings.jsm

```diff
@@ -0,0 +1,38 @@
+
+
+class FfiConverterString {
+    static lift(buf) {
+        return this.read(new ArrayBufferDataStream(buf));
```

Comment:
I think you just made the same mistake I made when I first looked at this.  If you look at the scaffolding code there's an asymmetry between `lift()`/`lower()` and `read()`/`write()`.  When we lift/lower, we skip reading/writing the length to the buffer because the buffer has it's own length field.

Reply:
yeah that makes sense, good catch... heh surprising it worked, but I think the `from_uf8_unchecked` + the fact that I just send it back on the Rust said made it seem as if it worked when it didn't. so the string in Rust is malformed (with the first 32 bits being the length), but since it just sends it back when we `lift` it in JS it seems as if it was fine 

wow... I'll fix it - really good catch, thanks @bendk!!

Reply:
I was very sensitive to this one, because it bit me very bad when I was first working on the string lifting/lowering code.